### PR TITLE
Switch to https to talk to API

### PR DESCRIPTION
**Pull request by hectorj2f**
It seems we are going to switch to HTTPS for the API endpoint as well as for other services. Once API switches to https (soon), we should also update our scripts.

### CODE REVIEW

**Comment by hectorj2f** [comment]
ping @pmuir. Do you approve it ?

File: README.adoc [hunk 1]
```diff
@@ -111,7 +111,7 @@ They're generally not meant to be run manually.
 To generate production build, set API URL and run build script as follows:
 
 ----
-$ export API_URL="http://api.example.org/api/"
+$ export API_URL="https://api.example.org/api/"
```
Comment (pmuir) [inline README.adoc]: We aren't building planner standalone in prod.

File: cico_build_deploy.sh [hunk 2]
```diff
@@ -27,7 +27,7 @@ service docker start
 cp /tmp/jenkins-env .
 docker build -t fabric8-planner-builder -f Dockerfile.builder .
 # User root is required to run webdriver-manager update. This shouldn't be a problem for CI containers
-mkdir -p dist && docker run --detach=true --name=fabric8-planner-builder --user=root --cap-add=SYS_ADMIN -e "API_URL=http://api.prod-preview.openshift.io/api/" -t -v $(pwd)/dist:/dist:Z fabric8-planner-builder
+mkdir -p dist && docker run --detach=true --name=fabric8-planner-builder --user=root --cap-add=SYS_ADMIN -e "API_URL=https://api.prod-preview.openshift.io/api/" -t -v $(pwd)/dist:/dist:Z fabric8-planner-builder
```
Comment (pmuir) [inline cico_build_deploy.sh]: No, for the prod build planner should not be using an env var to determine the api url

File: cico_build_deploy_npm.sh [hunk 3]
```diff
@@ -20,7 +20,7 @@ yum clean all
 service docker start
 
 docker build -t fabric8-planner-builder -f Dockerfile.builder .
-mkdir -p dist && docker run --detach=true --name=fabric8-planner-builder -e "API_URL=http://api.prod-preview.openshift.io/api/" -e JENKINS_URL -e GIT_BRANCH -e "CI=true" -e GH_TOKEN -e NPM_TOKEN -t -v $(pwd)/dist:/dist:Z fabric8-planner-builder
+mkdir -p dist && docker run --detach=true --name=fabric8-planner-builder -e "API_URL=https://api.prod-preview.openshift.io/api/" -e JENKINS_URL -e GIT_BRANCH -e "CI=true" -e GH_TOKEN -e NPM_TOKEN -t -v $(pwd)/dist:/dist:Z fabric8-planner-builder
```
Comment (pmuir) [inline cico_build_deploy_npm.sh]: No, for the NPM the api url should not be set, it is injected by the platform consuming planner.

File: cico_run_tests.sh [hunk 4]
```diff
@@ -29,7 +29,7 @@ service docker start
 cp /tmp/jenkins-env .
 docker build -t fabric8-planner-builder -f Dockerfile.builder .
 # User root is required to run webdriver-manager update. This shouldn't be a problem for CI containers
-mkdir -p dist && docker run --detach=true --name=fabric8-planner-builder --user=root --cap-add=SYS_ADMIN -e "API_URL=http://api.prod-preview.openshift.io/api/" -e "CI=true" -t -v $(pwd)/dist:/dist:Z fabric8-planner-builder
+mkdir -p dist && docker run --detach=true --name=fabric8-planner-builder --user=root --cap-add=SYS_ADMIN -e "API_URL=https://api.prod-preview.openshift.io/api/" -e "CI=true" -t -v $(pwd)/dist:/dist:Z fabric8-planner-builder
```
Comment (pmuir) [inline cico_run_tests.sh]: This is fine as it's for tests only.

File: config/webpack.prod.js [hunk 5]
```diff
@@ -21,7 +21,7 @@ const webpackMerge = require('webpack-merge');
  * Webpack Constants
  */
 const ENV = process.env.NODE_ENV = process.env.ENV = 'production';
-const API_URL = process.env.API_URL || 'http://api.almighty.io/api/';
+const API_URL = process.env.API_URL || 'https://api.almighty.io/api/';
```
Comment (pmuir) [inline config/webpack.prod.js]: No, in prod there shouldn't be an env var used.

File: cico_build_deploy_npm.sh (hunk 3 above)
Comment (sanbornsen) [inline cico_build_deploy_npm.sh]: I believe this file is itself not being used anywhere. So we can just remove this.

Comment (sanbornsen) [inline cico_build_deploy_npm.sh]: But maybe another commit and PR, since this PR is all about changing the http to https, we can keep this, can't we?

Comment (hectorj2f) [inline cico_build_deploy_npm.sh]: @sanbornsen it is used we need to keep it.. it is used in our CI pipelines

**Comment by hectorj2f** [comment]
@sanbornsen I see some references to `API_URL` in stats.json. That isn't used, right ?

**Comment by sanbornsen** [comment]
I have no idea what does **stats.json** file do. @joshuawilson can you help here?

**Comment by joshuawilson** [comment]
stats.json is a log used to compare the performance of the build. By tracking it in git we can see trends. This is not used by the running app. So if for some reason it is causing an issue, it can be removed.

**Comment by sanbornsen** [comment]
@hectorj2f  so we are good to go.

**Comment by hectorj2f** [comment]
@sanbornsen yes!

**Comment by hectorj2f** [comment]
Thanks for the review @pmuir 👍

**Pull request merged**